app/models/record.rb

```diff
@@ -28,11 +28,22 @@ class Record < ApplicationRecord
   scope :visible, -> { where.not(state: Record::STATE_HIDDEN) }
   scope :metadata_format, ->(format) { where(format: format) }
 
+  after_update :update_associated_granule_state
+
+  def update_associated_granule_state
+    unless previous_changes['state'].nil?
+      if recordable_type == 'Collection' && has_associated_granule?(self)
+        granule_record = Record.find_by id: self.associated_granule_value
+        granule_record.state = self.state
+        granule_record.save
```

Comment:
Isn't this going to skip the AASM validations?

Reply:
Yes, but only for the granule side.   The validations are now done by calling the methods directly on the record model object during association or changing the state of the collection.  This also provides slightly better and more specific error messages since the validation is being done on two different records at the same time.